Library/Formula/berkeley-db.rb

```diff
@@ -33,6 +33,20 @@ def install
       --enable-cxx
       --enable-compat185
     ]
+
+    if build.with? "java"
+      java_home = ENV["JAVA_HOME"] = `/usr/libexec/java_home`.chomp
+
+      # The Oracle JDK puts jni.h into #{java_home}/include and jni_md.h into
+      # #{java_home}/include/darwin.  The original Apple/SUN JDK placed jni.h
+      # and jni_md.h into
+      # /System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/
+      #
+      # Setup the include path with the Oracle JDK location first and the Apple JDK location second.
+      ENV["CFLAGS"] = "-I#{java_home}/include" <<
+                      " -I#{java_home}/include/darwin" <<
+                      " -I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/"
+    end
```

Comment:
Pretty sure `:java` already checks for the JDK. That seems sensible. `Requirements` handle superenv by use of the `env` method.